integration/csi/main.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/Alluxio/alluxio-csi/alluxio"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+)
+
+var (
+	endpoint string
+	nodeID   string
+)
+
+func init() {
+	flag.Set("logtostderr", "true")
+}
+
+func main() {
+
+	flag.CommandLine.Parse([]string{})
+
+	cmd := &cobra.Command{
+		Use:   "Alluxio",
+		Short: "CSI based Alluxio driver",
+		Run: func(cmd *cobra.Command, args []string) {
+			handle()
+		},
+	}
+
+	cmd.Flags().AddGoFlagSet(flag.CommandLine)
+
+	cmd.PersistentFlags().StringVar(&nodeID, "nodeid", "", "node id")
+	cmd.MarkPersistentFlagRequired("nodeid")
+
+	cmd.PersistentFlags().StringVar(&endpoint, "endpoint", "", "CSI endpoint")
+	cmd.MarkPersistentFlagRequired("endpoint")
+
+	cmd.ParseFlags(os.Args[1:])
+	if err := cmd.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "%s", err.Error())
+		os.Exit(1)
+	}
+
+	os.Exit(0)
+}
+
+func handle() {
+	startReaper()
+
+	d := alluxio.NewDriver(nodeID, endpoint)
+	d.Run()
+}
+
+/*
+Based on https://github.com/openshift/origin/blob/master/pkg/util/proc/reaper.go
```

Comment:
this link cannot be opened, try using a permanent link instead of the master branch link